### PR TITLE
Test for function calculatorCoordinateOfTopLeft. Issue #9

### PR DESCRIPTION
**Pull request by PashaBeshchuk**

### CODE REVIEW

File: test/test-calculatorCoordinateOfTopLeft .js [hunk 1]
```diff
@@ -1,2 +1,81 @@
...
+    it("Arrays is many elements", function(){
+        arrayOfCoordinates = tetris.getCoordinates(`
+            -----
+            -----
+            ----X
+            -----
+            ---X-
+        `)
+        let topLeftCorner = calculatorCoordinateOfTopLeft(arrayOfCoordinates)
+        expectedResult = tetris.getCoordinates(`
```
Comment (zbstof) [inline test/test-calculatorCoordinateOfTopLeft .js]: why this coordinate?

File: test/test-calculatorCoordinateOfTopLeft .js [hunk 2]
```diff
@@ -1,2 +1,81 @@
...
+            X----
```
Comment (zbstof) [inline test/test-calculatorCoordinateOfTopLeft .js]: why this coordinate?

File: test/test-calculatorCoordinateOfTopLeft .js [hunk 3]
```diff
@@ -1,2 +1,81 @@
...
+            -----
+            --XX-
+            ---X-
+            ---X-
+            -----
+        `)
+        let topLeftCorner = calculatorCoordinateOfTopLeft(arrayOfCoordinates)
+        expectedResult = tetris.getCoordinates(`
+            -----
+            -X---
```
Comment (zbstof) [inline test/test-calculatorCoordinateOfTopLeft .js]: why this coordinate?

File: test/test-calculatorCoordinateOfTopLeft .js (hunk 2 above)
Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: Я посмотрел по какой траектории движутся элементы и выбрал верхнюю левую точку

File: test/test-calculatorCoordinateOfTopLeft .js (hunk 1 above)
Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: Два элемента которые не находятся рядом, потому их выбрал, это может быть J-тетрамино, только без точек между хвостом и головой

File: test/test-calculatorCoordinateOfTopLeft .js [hunk 4]
```diff
@@ -1,2 +1,81 @@
...
+            -------
+            -------
+            -------
+            ------X
+            -------
+            
+        `)
+        let topLeftCorner = calculatorCoordinateOfTopLeft(arrayOfCoordinates)
+        expectedResult = tetris.getCoordinates(`
+            X------
```
[the remark this answers is not in the document]
Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: верхняя левая точка, по оси которой двигается элемент

**Comment by zbstof** [comment]
Where in the requirements does it say that any of those elements are moving?

File: test/test-calculatorCoordinateOfTopLeft .js (hunk 4 above)
[the remark this answers is not in the document]
Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: то есть, верхняя левая точка при данном положении тетрамино, верно?

[the remark this answers is not in the document]
Comment (zbstof) [reply test/test-calculatorCoordinateOfTopLeft .js]: What tetromino are you talking about? arrayOfCoordinates is one coordinate

File: test/test-calculatorCoordinateOfTopLeft .js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            -----
+            --X--
+            -----
+            -----
+            -----
+        `)
+    })
+})
+
+describe("Coordinates spaced apart, is 3 elements, does not match coordinate of any of the elements", function(){
```
Comment (zbstof) [inline test/test-calculatorCoordinateOfTopLeft .js]: Why do you have one "it" block per "describe" block?
Why all "it" blocks are named "Arrays is many elements"?

Comment (zbstof) [reply test/test-calculatorCoordinateOfTopLeft .js]: Please leave one describe and rename it blocks

Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: Sorry,  I don t understand you, what do you mean "Where are assert statements?"?

Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: Can I merge?

**Comment by zbstof** [comment]
Please fox all comments

File: test/test-calculatorCoordinateOfTopLeft .js (hunk 5 above)
Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: I checked everything, I have done everything

File: test/test-calculatorCoordinateOfTopLeft .js [hunk 6]
```diff
@@ -1,2 +1,81 @@
...
+        expectedResult = tetris.getCoordinates(`
+            X----
+            -----
+            -----
+            -----
+            -----
+        `)
+})
+
+describe("Coordinates spaced apart, is 2 elements, does not match coordinate of any of the elements", function(){
```
[the remark this answers is not in the document]
Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: Done

File: test/test-calculatorCoordinateOfTopLeft .js (hunk 4 above)
[the remark this answers is not in the document]
Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: Done

File: test/test-calculatorCoordinateOfTopLeft .js (hunk 5 above)
Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: Done
"These are good.
Please add test where top left does not match any of the coordinates, like z-tetromino
Edit: i mean s-tetromino"

Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: Done 
"Where are assert statements?"

Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: 1. All of these are incorrect
Add a test for square O-tetromino first - you should have no problems locating top left corner
DONE

2. why this coordinate?
DONE

3. you have single element, why the test sais otherwise?
DONE

4. why this coordinate?
    What tetromino are you talking about? arrayOfCoordinates is one coordinate
DONE	

5. These are good.
   Please add test where top left does not match any of the coordinates, like z-tetromino
   Edit: i mean s-
DONE

6. First element is matching left corner, please change
DONE
7. Why do you have spaces at the end of filename? Please remove
DONE
8. Why do you have one "it" block per "describe" block?
    Why all "it" blocks are named "Arrays is many elements"?
DONE

9. Please leave one describe and rename it blocks
DONE
10. Where are assert statements?
DONE
11. Please fix all comments
Maybe done

**Comment by zbstof** [comment]
> Why do you have spaces at the end of filename? Please remove
> DONE
In which commit did you do it?

**Comment by zbstof** [comment]
There should not be spaces before file extension

File: test/test-calculatorCoordinateOfTopLeft .js (hunk 5 above)
Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: "Why do you have spaces at the end of filename? Please remove"
I understand it and did this.
I forgot push this, sorry...
Can I merge?

Comment (PashaBeshchuk) [reply test/test-calculatorCoordinateOfTopLeft .js]: In the last
"In which commit did you do it?"
